src/lightning/pytorch/loops/utilities.py

```diff
@@ -156,23 +152,7 @@ def _decorator(self: _Loop, *args: Any, **kwargs: Any) -> Any:
         if not hasattr(self, "inference_mode"):
             raise TypeError(f"`{type(self).__name__}.inference_mode` needs to be defined")
         context_manager: Type[ContextManager]
-        if dist.is_available() and dist.is_initialized() and dist.get_backend() == "gloo":
-            # gloo backend does not work properly.
-            # https://github.com/Lightning-AI/lightning/pull/12715/files#r854569110
-            # TODO: explore why and possibly open an issue in PyTorch repository
-            context_manager = torch.no_grad
-        elif isinstance(self.trainer.accelerator, TPUAccelerator):
-            context_manager = torch.no_grad
-        elif _TORCH_GREATER_EQUAL_1_13 and isinstance(self.trainer.strategy, FSDPStrategy):
-            # https://github.com/pytorch/pytorch/issues/95957
-            context_manager = torch.no_grad
-        elif _TORCH_EQUAL_2_0 and self.trainer.lightning_module._compiler_ctx is not None:
-            # avoid: `RuntimeError: Inference tensors do not track version counter` fixed in v2.1
-            context_manager = torch.no_grad
-        elif self.inference_mode:
-            context_manager = torch.inference_mode
-        else:
-            context_manager = torch.no_grad
```

Comment:
What happened to this? This needs to stay

Reply:
why do you need to stay when you effectively set `torch.inference_mode` only for one case and the rest is `torch.no_grad`?

Reply:
It's chosen based on all those removed conditions. There will be failing tests because of this change

Reply:
ok, I'll revisit it

Reply:
reverted